tests/unit/test_zilien_reader.py

```diff
@@ -0,0 +1,385 @@
+from pathlib import Path
+from unittest.mock import patch
+
+import numpy as np
+import pytest
+
+from ixdat.techniques import ECMeasurement, MSMeasurement, ECMSMeasurement
+from ixdat.readers.zilien import ZilienTSVReader, to_mass, parse_metadata_line
+from ixdat.data_series import TimeSeries, ValueSeries
+
+
+DATA_DIR = Path(__file__).parent.parent.parent / "submodules" / "ixdat-large-test-files"
+
+
+def patch_biologic_dataset_part(column_headers_cut, data_cut):
+    """A patch function to replace the original one inside the _form_series method."""
+    return [
+        (column_header, data_cut[0, i])
+        for i, column_header in enumerate(column_headers_cut)
+    ]
+
+
+def patch_zilien_dataset_part(series_header, column_headers_cut, data_cut):
+    """A patch function to replace the original one inside the _form_series method."""
+    mass = to_mass(series_header)
+    aliases = {f"M{mass}": [f"M{mass} [A]"]} if mass is not None else {}
+
+    return [
+        (f"{series_header}_{column_header}", data_cut[0, i])
+        for i, column_header in enumerate(column_headers_cut)
+    ], aliases
+
+
+class TestZilienTSVReader:
+    """Tests for Zilien TSV Reader."""
+
+    def test_create_series_objects(self):
+        """Test creation of Ixdat series objects."""
+
+        reader = ZilienTSVReader()
+        reader._timestamp = 123456
+
+        column_headers = ["time/s", "experiment_number", "I/mA", "Ewe/V"]
+        names_and_units = [
+            ("Biologic time/s", "s", None),
+            ("experiment_number", "", None),
+            ("I/mA", "mA", None),
+            ("Ewe/V", "V", None),
+        ]
+        data = np.array(
+            [
+                [0.1, 1, 3, "NaN"],
+                [0.2, 1, 4, "NaN"],
+            ],
+            dtype=float,
+        )
+
+        series = reader._create_series_objects(column_headers, names_and_units, data)
```

Comment:
I don't see the logic of this function - why it needs four column headers and four tuples of names_and_units to make two series.

Reply:
Beside testing the series creation, it is also testing that `experiment_number` and `technique_number` columns, and columns with only "NaN" values are skipped.